tools/souper-check.cpp

```diff
@@ -70,15 +74,34 @@ int SolveInst(const MemoryBufferRef &MB, Solver *S) {
   }
 
   unsigned Index = 0;
+  int Ret = 0;
+  int Success = 0, Fail = 0, Error = 0;
   for (auto Rep : Reps) {
-    if (InferRHS) {
+    if (InferRHS || ReInferRHS) {
+      int OldCost;
+      if (ReInferRHS) {
+        OldCost = cost(Rep.Mapping.RHS);
+        Rep.Mapping.RHS = 0;
+      }
       if (std::error_code EC = S->infer(Rep.BPCs, Rep.PCs, Rep.Mapping.LHS,
                                         Rep.Mapping.RHS, IC)) {
         llvm::errs() << EC.message() << '\n';
-        return 1;
+        Ret = 1;
+        ++Error;
       }
       if (Rep.Mapping.RHS) {
-        llvm::outs() << "; RHS inferred successfully\n";
+        ++Success;
+        if (ReInferRHS) {
+          int NewCost = cost(Rep.Mapping.RHS);
+          if (NewCost <= OldCost)
+            llvm::outs() << "; RHS inferred successfully, no cost regression (Old= " <<
+              OldCost << ", New= " << NewCost << ")\n";
+          else
```

Comment:
We will need to enable ``-souper-synthesis-ignore-cost`` per default, otherwise this branch will be never executed.

Reply:
I see what you mean.
I need to look into how the cost function plays into synthesis, will report back.